driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -594,10 +594,10 @@ public boolean onTimeout(Connection connection, long latency, int retryCount) {
         }
 
         Host queriedHost = current;
+        // If a query times out, we consider that the host is unstable, so we defunct
+        // the connection to mark it down.
+        OperationTimedOutException timeoutException = new OperationTimedOutException(connection.address);
```

Comment:
There was an OperationTimedOutException already, but curiously enough, it wasn't used. I think this is the right place for it. I discovered that while creating subclasses of DriverException within JAVA-698.